src/bridge/cockpitconnect.c

```diff
@@ -247,37 +242,39 @@ cockpit_connect_stream_full (CockpitConnectable *connectable,
                              GAsyncReadyCallback callback,
                              gpointer user_data)
 {
-  GSimpleAsyncResult *simple;
+  GTask *task;
   ConnectStream *cs;
 
   g_return_if_fail (connectable != NULL);
   g_return_if_fail (G_IS_SOCKET_CONNECTABLE (connectable->address));
   g_return_if_fail (!cancellable || G_IS_CANCELLABLE (cancellable));
 
-  simple = g_simple_async_result_new (NULL, callback, user_data, cockpit_connect_stream);
+  task = g_task_new (NULL, NULL, callback, user_data);
+  g_task_set_source_tag (task, cockpit_connect_stream); /* FIXME: do we need this? */
   cs = g_new0 (ConnectStream, 1);
   cs->connectable = cockpit_connectable_ref (connectable);
   cs->cancellable = cancellable ? g_object_ref (cancellable) : NULL;
   cs->enumerator = g_socket_connectable_enumerate (connectable->address);
-  g_simple_async_result_set_op_res_gpointer (simple, cs, connect_stream_free);
+  g_message ("XXX cockpit_connect_stream_full set task %p ptr %p", task, cs);
+  g_task_return_pointer (task, cs, connect_stream_free);
```

Comment:
That's your issue: `g_task_return_pointer()` immediately completes the task.  `g_simple_async_result_set_op_res_gpointer()` doesn't do that without a corresponding clal to `g_simple_async_result_complete()`.  Use `g_task_set_task_data()` instead.